File: C++/4.3.3.cpp
#include <iostream>
#include <vector>
#include <iomanip>
#include <limits>

using namespace std;

const int SIZE = 5;

void InputMatrix(vector<vector<double>>& M) {
    cout << "Enter the elements of the matrix A:" << endl;
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            cout << "Enter element A[" << i + 1 << "," << j + 1 << "]: ";
            cin >> M[i][j];
        }
    }
}

void FindMaxRowSum(const vector<vector<double>>& M, int& maxIndex, double& maxSum) {
    maxSum = numeric_limits<double>::lowest();
    for (int i = 0; i < SIZE; i++) {
        double sum = 0;
        for (int j = 0; j < SIZE; j++) {
            sum += M[i][j];
        }
        if (sum > maxSum) {
            maxSum = sum;
            maxIndex = i;
        }
    }
}

void FindMaxColSum(const vector<vector<double>>& M, int& maxIndex, double& maxSum) {
    maxSum = numeric_limits<double>::lowest();
    for (int j = 0; j < SIZE; j++) {
        double sum = 0;
        for (int i = 0; i < SIZE; i++) {
            sum += M[i][j];
        }
        if (sum > maxSum) {
            maxSum = sum;
            maxIndex = j;
        }
    }
}

void CalculateSum(const vector<vector<double>>& M, int rowIndex, int colIndex, double& sum) {
    sum = 0;
    for (int i = 0; i < SIZE; i++) {
        sum += M[rowIndex][i] + M[i][colIndex];
    }
    // Убираем двойной счет элемента на пересечении строки и столбца
    sum -= M[rowIndex][colIndex];
}

int main() {
    vector<vector<double>> A(SIZE, vector<double>(SIZE));
    int maxRowIndex = 0, maxColIndex = 0;
    double maxRowSum = 0, maxColSum = 0;

    // Вводим данные для матрицы
    InputMatrix(A);

    // Выводим матрицу после ввода данных
    cout << "\nMatrix A after input:" << endl;
    for (int i = 0; i < SIZE; i++) {
        for (int j = 0; j < SIZE; j++) {
            cout << fixed << setprecision(2) << A[i][j] << " ";
        }
        cout << endl;
    }

    // Находим строку с максимальной суммой элементов
    FindMaxRowSum(A, maxRowIndex, maxRowSum);

    // Находим столбец с максимальной суммой элементов
    FindMaxColSum(A, maxColIndex, maxColSum);

    // Проверяем, совпадают ли номера строки и столбца
    if (maxRowIndex == maxColIndex) {
        // Вычисляем сумму элементов строки и столбца
        CalculateSum(A, maxRowIndex, maxColIndex, maxRowSum);
        cout << "Sum of the elements of the row and column: " << fixed << setprecision(2) << maxRowSum << endl;
    } else {
        // Выводим номера строки и столбца с максимальными суммами
        cout << "Row with the maximum sum: " << maxRowIndex + 1 << endl;
        cout << "Column with the maximum sum: " << maxColIndex + 1 << endl;
    }

    return 0;
}

File: pascal/4.3.3.pas
const
  SIZE = 5;

type
  TMatrix = array[1..SIZE, 1..SIZE] of Real;

var
  A: TMatrix;
  i, j: Integer;
  maxRowIndex, maxColIndex: Integer;
  maxRowSum, maxColSum: Real;

procedure InputMatrix(var M: TMatrix);
begin
  WriteLn('Enter elements of matrix:');
  for i := 1 to SIZE do
    for j := 1 to SIZE do
    begin
      Write('Enter element A[', i, ',', j, ']: ');
      ReadLn(M[i, j]);
    end;
end;

procedure FindMaxRowSum(const M: TMatrix; var maxIndex: Integer; var maxSum: Real);
var
  i, j: Integer;
  sum: Real;
begin
  maxSum := -MaxInt;
  for i := 1 to SIZE do
  begin
    sum := 0;
    for j := 1 to SIZE do
      sum := sum + M[i, j];
    if sum > maxSum then
    begin
      maxSum := sum;
      maxIndex := i;
    end;
  end;
end;

procedure FindMaxColSum(const M: TMatrix; var maxIndex: Integer; var maxSum: Real);
var
  i, j: Integer;
  sum: Real;
begin
  maxSum := -MaxInt;
  for j := 1 to SIZE do
  begin
    sum := 0;
    for i := 1 to SIZE do
      sum := sum + M[i, j];
    if sum > maxSum then
    begin
      maxSum := sum;
      maxIndex := j;
    end;
  end;
end;

procedure CalculateSum(const M: TMatrix; rowIndex, colIndex: Integer; var sum: Real);
var
  i: Integer;
begin
  sum := 0;
  for i := 1 to SIZE do
  begin
    sum := sum + M[rowIndex, i] + M[i, colIndex];
  end;
  // Убираем двойной счет элемента на пересечении строки и столбца
  sum := sum - M[rowIndex, colIndex];
end;

begin
  // Вводим данные для матрицы
  InputMatrix(A);

  // Находим строку с максимальной суммой элементов
  FindMaxRowSum(A, maxRowIndex, maxRowSum);

  // Находим столбец с максимальной суммой элементов
  FindMaxColSum(A, maxColIndex, maxColSum);

  // Проверяем, совпадают ли номера строки и столбца
  if maxRowIndex = maxColIndex then
  begin
    // Вычисляем сумму элементов строки и столбца
    CalculateSum(A, maxRowIndex, maxColIndex, maxRowSum);
    WriteLn('Sum of row and column elements: ', maxRowSum:0:2);
  end
  else
  begin
    // Выводим номера строки и столбца с максимальными суммами
    WriteLn('Row number with maximum amount: ', maxRowIndex);
    WriteLn('Column number with maximum sum: ', maxColIndex);
  end;
end.